execution/executor/src/lib.rs

```diff
@@ -503,6 +503,7 @@ where
             Err(anyhow::anyhow!("Injected error in execute_chunk"))
         });
 
+        let transaction_outputs_clone = transaction_outputs.clone();
```

Comment:
Aah, annoying -- missed the borrow the first time. To avoid cloning let's just find out if they were provided here?
e.g.: `let transactions_executed = transaction_outputs.is_none()`